src/snowflake/snowpark/_internal/utils.py

```diff
@@ -791,3 +791,29 @@ def parse_table_name(table_name: str) -> List[str]:
 
     ret.append(table_name[cur_word_start_idx:i])
     return ret
+
+
+EMPTY_STRING = ""
+DOUBLE_QUOTE = '"'
+ALREADY_QUOTED = re.compile('^(".+")$')
+UNQUOTED_CASE_INSENSITIVE = re.compile("^([_A-Za-z]+[_A-Za-z0-9$]*)$")
+
+
+def quote_name(name: str) -> str:
+    if ALREADY_QUOTED.match(name):
```

Comment:
maybe add docstring to this function? Also, is it a good idea to have this ALREADY_QUOTED check here? It's a bit confusing, because name could be 'A' but maye require additional quoting. Ideally this function should have a 1:1 mapping I feel.